components/cam/src/physics/cam/micro_p3_interface.F90

```diff
@@ -0,0 +1,1555 @@
+    call pbuf_get_field(pbuf,  cv_reffliq_idx, cvreffliq,    col_type=col_type)
+    call pbuf_get_field(pbuf,  cv_reffice_idx, cvreffice,    col_type=col_type)
+    call pbuf_get_field(pbuf, cc_t_idx,        cc_t,     start=(/1,1,itim_old/), kount=(/psetcols,pver,1/), col_type=col_type)
+    call pbuf_get_field(pbuf, cc_qv_idx,       cc_qv,    start=(/1,1,itim_old/), kount=(/psetcols,pver,1/), col_type=col_type)
+    call pbuf_get_field(pbuf, cc_ql_idx,       cc_ql,    start=(/1,1,itim_old/), kount=(/psetcols,pver,1/), col_type=col_type)
+    call pbuf_get_field(pbuf, cc_qi_idx,       cc_qi,    start=(/1,1,itim_old/), kount=(/psetcols,pver,1/), col_type=col_type)
+    call pbuf_get_field(pbuf, cc_nl_idx,       cc_nl,    start=(/1,1,itim_old/), kount=(/psetcols,pver,1/), col_type=col_type)
+    call pbuf_get_field(pbuf, cc_ni_idx,       cc_ni,    start=(/1,1,itim_old/), kount=(/psetcols,pver,1/), col_type=col_type)
+    call pbuf_get_field(pbuf, cc_qlst_idx,     cc_qlst,  start=(/1,1,itim_old/), kount=(/psetcols,pver,1/), col_type=col_type)
+    !call pbuf_get_field(pbuf,   tnd_qsnow_idx, tnd_qsnow,    col_type=col_type) 
+    !call pbuf_get_field(pbuf,   tnd_nsnow_idx, tnd_nsnow,    col_type=col_type) 
+    !call pbuf_get_field(pbuf,      re_ice_idx,    re_ice,    col_type=col_type) 
+    call pbuf_get_field(pbuf,     acprecl_idx,   acprecl                       ) 
+    call pbuf_get_field(pbuf,      acgcme_idx,    acgcme                       ) 
+    call pbuf_get_field(pbuf,       acnum_idx,     acnum                       ) 
+    !call pbuf_get_field(pbuf,       qrain_idx,     qrain,    col_type=col_type) 
+    !call pbuf_get_field(pbuf,       qsnow_idx,     qsnow,    col_type=col_type) 
+    !call pbuf_get_field(pbuf,       nrain_idx,     nrain,    col_type=col_type) 
+    !call pbuf_get_field(pbuf,       nsnow_idx,     nsnow,    col_type=col_type) 
+
+    
+    rd = rair  ! Probably not needed now that we are using Exner
+    cp = cpair ! Probably not needed now that we are using Exner
+    inv_cp = 1./cp ! Dido from above
+    
+    ncol = state%ncol
+    !==============
+    ! Some pre-microphysics INITIALIZATION
+    !==============
+    cldo(:ncol,top_lev:pver)=ast(:ncol,top_lev:pver)
+    ! INITIALIZE PTEND
+    !==============
+    !ptend is an output variable. Since not substepping in micro, don't need 
+    !a local copy.
+
+    lq            = .false. !initialize all constituents to false.
+    lq(1)         = .true.
+    lq(ixcldliq)  = .true.
+    lq(ixcldice)  = .true.
+    lq(ixnumliq)  = .true.
+    lq(ixnumice)  = .true.
+    lq(ixrain)    = .true.
+    lq(ixcldrim)  = .true.
+    lq(ixnumrain) = .true.
+    lq(ixrimvol)  = .true.
+
+    call physics_ptend_init(ptend, psetcols, "micro_p3", ls=.true., lq=lq)
+
+    ! HANDLE AEROSOL ACTIVATION
+    !==============
+    !saving this for later... use prescribed Nd for now.
+    log_predictNc = .false.
```

Comment:
Do you want log_predictNc hardwired to false in this PR? I'm ok with leaving this way for now as I will make it a namelist option with my upcoming PR.

Reply:
Yes, I think that is a good plan forward since this PR represents the initial port of P3 and later PR's will incorporate improvements.